### DISCUSSION TITLE
Populate int column from relationship (and perform math on it)

### DISCUSSION THREAD

**Discussion by KristofferBroberg** [Budibase/budibase#9775] [Help]
Hi all!

I´ve got two tables

1.
_Status_ / _int_

2.

_statuses_ /  _values_

There's a working relationship between _Status_ in 1 and _statuses_ in 2. Now I want to populate _int_ in 1 with data from _values_ in 2. This can be done using the formula column but then I'm not able to perform math on it in the automations (right?), which is required. Any ideas on how I could solve this?

I'm using the internal db. Thankful for any advice!

**Comment by melohagan** [comment]
Hey @KristofferBroberg 

How is your automation triggered? If you use the **App action** trigger then you could pass through a value that is calculated in a front-end JavaScript binding.

Some screenshots detailing what you are looking to do could also be useful!

**Reply by KristofferBroberg** [reply]
It´s triggered by an app action. Basically what I'm wondering is  this:

In the internal db there´s an int column. The value is...say, 4. Via app action, is there a way to subtract 1 from this column? In the db, not just front end.

If yes, is the above still possible if that column is a formula column?

Forgive me for my very rudimentary understanding of how databases and Budibase work!

**Reply by melohagan** [reply]
If you are using an app action, then you could pass in the calculated value from the front-end:

![Screenshot 2023-03-03 at 09 29 32](https://user-images.githubusercontent.com/101575380/222683905-ad1a0cf7-7a36-4fbf-951a-9cf3a29e2e16.png)

![Screenshot 2023-03-03 at 09 31 55](https://user-images.githubusercontent.com/101575380/222684405-e4a5cb42-8861-4e36-abdd-c83c0df67d4b.png)

![Screenshot 2023-03-03 at 09 31 23](https://user-images.githubusercontent.com/101575380/222684292-c8dfda09-5b76-4c22-bd43-8205751734e9.png)

**Reply by KristofferBroberg** [reply]
Ok I see. Is there any way to, instead of {{ minus 4 1 }}, go {{ minus x 1}} ... meaning doing the subtraction without knowing the actual value? I´ve looked throught the binding/handelbars docs without finding an answer. Seems so basic that it should be fairly straight forward.

**Reply by melohagan** [reply]
The *x* would be the binding for the relationship field, which depending on your component tree would be obtained from a [repeater](https://docs.budibase.com/docs/fetching-data#displaying-relational-data), or from a [formula](https://docs.budibase.com/docs/formula#fields-from-relationships) binding

**Reply by KristofferBroberg** [reply]
OK. Not sure I fully grasp what you´re saying but I'm gonna be a good soldier and study the docs. and do some trial and error. The binding part is a bit confusing!

**Reply by KristofferBroberg** [reply]
Thank you!
